### PR TITLE
[dev-client] Partially fixed retrieving the React Context from the Android Flipper plugin

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-5413.

# How

Applied the `expo-dev-launcher` part of Kudo's proposed solution - https://exponent-internal.slack.com/archives/C011V63429H/p1655385628638509.
Added a new point to the troubleshooting page that describes the second part which has to be applied to the main application.

# Test Plan

Confirming that this workaround works ✅

### CODE REVIEW

File: docs/pages/development/troubleshooting.md [hunk 1]
```diff
@@ -27,3 +29,9 @@ you have an outdated version of `expo-modules-autolinking` (below 0.7.x) install
 A common reason for this is installing `expo-cli` or `eas-cli` as dependencies of your project, rather than globally as we recommend; removing those packages should fix the error.
 
 Otherwise, run `yarn why expo-modules-autolinking` to see where the outdated version is coming from, and upgrade that package to the latest version.
+
+### Cannot retrieve the React Context from the Android Flipper plugin
```
Comment (esamelson) [inline docs/pages/development/troubleshooting.md:33]: Is this the exact error message as logged? If not, it might be helpful to include that so people can find this page by searching the error message.

**Review by esamelson — COMMENTED** [review]
Thanks for taking care of this, it LGTM. What happens if you do include this workaround in the dev-launcher, but don't make the change to `ReactNativeFlipper.java`? (Just want to make sure that the new `maybeInitFlipper` method doesn't make things _worse_ without the changes in `ReactNativeFlipper`.)

**Comment by lukmccall** [comment]
> What happens if you do include this workaround in the dev-launcher, but don't make the change to ReactNativeFlipper.java?

It wasn't a problem. However, I've changed the code a little bit. Right now, this fix doesn't require changes in the `ReactNativeFlipper.java`.

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoader.kt [hunk 2]
```diff
@@ -94,4 +96,59 @@ abstract class DevLauncherAppLoader(
...
+    } catch (e: IllegalAccessException) {
+      e.printStackTrace()
+    } catch (e: InvocationTargetException) {
+      e.printStackTrace()
+    }
+  }
+
+  private fun tryToStopFlipper(): Boolean {
+    val androidFlipperClientClass = Class.forName("com.facebook.flipper.android.AndroidFlipperClient");
+    val getInstanceMethod = androidFlipperClientClass.getMethod("getInstanceIfInitialized")
```
Comment (esamelson) [inline packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoader.kt:131]: Do you think it's worth adding a separate try-catch block around here, or elsewhere in this method, so that if these method/field names change in the future we will still reach the `initializeFlipper` call?

File: docs/pages/development/troubleshooting.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 title: Troubleshooting
 ---
 
+import ConfigurationDiff from '~/components/plugins/ConfigurationDiff';
```
Comment (esamelson) [inline docs/pages/development/troubleshooting.md:5]: can revert the changes to this file

**Review by esamelson — APPROVED** [review]
Awesome, this is a great improvement over developers needing to modify their own ReactNativeFlipper class as well.

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoader.kt (hunk 2 above)
Comment (lukmccall) [reply packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/launcher/loaders/DevLauncherAppLoader.kt:131]: I think we should disable this workaround when one of the methods in that method fails. Staring a flipper when one instance is already running doesn't have any effect, but what if the flipper was stope but plugins weren't removed. It's hard to predict what will happen. By not adding the `try-catch` block here, we will only use a workaround when we're certain that it will work.
